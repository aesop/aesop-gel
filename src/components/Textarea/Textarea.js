import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './Textarea.module.css';

const Textarea = ({
  className,
  dataTestRef,
  defaultValue,
  id,
  isEnabled,
  inputRef,
  maxLength,
  name,
  onBlur,
  onChange,
  rows,
  textareaClassName,
  theme,
  value,
}) => {
  const baseClassSet = cx(styles.base, styles[theme], className);
  const textareaClassSet = cx(
    styles.textarea,
    styles[theme],
    textareaClassName,
  );
  return (
    <label htmlFor={id}>
      <div className={baseClassSet}>
        <textarea
          className={textareaClassSet}
          data-test-ref={dataTestRef}
          defaultValue={defaultValue}
          disabled={!isEnabled}
          id={id}
          maxLength={maxLength}
          name={name}
          onBlur={onBlur}
          onChange={onChange}
          ref={inputRef}
          rows={rows}
          value={value}
        />
      </div>
    </label>
  );
};

Textarea.propTypes = {
  className: PropTypes.string,
  dataTestRef: PropTypes.string,
  defaultValue: PropTypes.string,
  id: PropTypes.string.isRequired,
  isEnabled: PropTypes.bool,
  inputRef: PropTypes.func,
  maxLength: PropTypes.number,
  name: PropTypes.string,
  textareaClassName: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  rows: PropTypes.number,
  theme: PropTypes.oneOf(['dark', 'light']),
  value: PropTypes.string,
};

Textarea.defaultProps = {
  className: undefined,
  dataTestRef: undefined,
  defaultValue: undefined,
  id: undefined,
  isEnabled: true,
  inputRef: undefined,
  maxLength: undefined,
  name: undefined,
  textareaClassName: undefined,
  onBlur: undefined,
  onChange: undefined,
  rows: undefined,
  theme: 'dark',
  value: undefined,
};

export { Textarea };
