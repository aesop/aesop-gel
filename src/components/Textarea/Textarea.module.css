@import 'styles/index';

.base {
  display: inline-block;
  width: 100%;
  box-sizing: border-box;
  padding-top: 16px;
  padding-right: 7px;
  padding-bottom: 18px;
  padding-left: 14px;
  border: 1px solid var(--color-dark-copy);
  margin-top: 23px;
  border-radius: 3px;
  &.dark {
    border-color: var(--color-dark-copy);
  }

  &.light {
    border-color: var(--color-light-copy);
  }
}

.textarea {
  width: 100%;
  height: 90px;
  border: none;
  background: transparent;
  color: var(--color-dark-copy);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(12px);
  line-height: 1.67;
  outline: none;
  resize: none;
  &.dark {
    color: var(--color-dark-copy);
  }

  &.light {
    color: var(--color-light-copy);
  }

  &::-webkit-scrollbar {
    width: 5px;
    margin-right: -10px;
    -webkit-appearance: none;
  }
  &::-webkit-scrollbar-thumb {
    border-radius: 4px;
    -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
  }

  &.dark::-webkit-scrollbar-thumb {
    background-color: var(--color-dark-copy);
  }
  &.light::-webkit-scrollbar-thumb {
    background-color: var(--color-light-copy);
  }
}
