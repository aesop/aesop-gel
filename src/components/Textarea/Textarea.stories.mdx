import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import Podium from '~/components/Podium';
import Textarea from './Textarea';

# Textarea

<Meta title="Components/Textarea" component={Textarea} />

<Preview>
  <Story name="Base component">
    <Podium horizontalPadding="small" verticalPadding="small">
      <Textarea
        id='test-textarea'
        theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
      />
    </Podium>
  </Story>
</Preview>

## API

<Props of={ContentHubArticle} />

