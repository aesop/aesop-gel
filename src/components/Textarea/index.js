/* istanbul ignore file */
import { Textarea } from './Textarea';

export { Textarea };
