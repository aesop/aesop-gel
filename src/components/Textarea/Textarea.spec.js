import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Textarea from './Textarea';

configure({ adapter: new Adapter() });

describe('<Textarea />', () => {
  it('should be defined', () => {
    expect(Textarea).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer.create(<Textarea id="test-textarea" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
