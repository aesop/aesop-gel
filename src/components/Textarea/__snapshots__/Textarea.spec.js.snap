// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> renders base component correctly 1`] = `
<label
  htmlFor="test-textarea"
>
  <div
    className="base dark"
  >
    <textarea
      className="textarea dark"
      data-test-ref="test-data-ref"
      disabled={false}
      id="test-textarea"
    />
  </div>
</label>
`;
